### PR TITLE
Add release notes for 2.8.4

### PR DESCRIPTION
**Pull request by serinamarie**
[Rendered](https://github.com/PrefectHQ/prefect/blob/release-notes-2.8.4/RELEASE-NOTES.md#release-284)

### Checklist

- [ ] This pull request references any related issue by including "closes `<link to issue>`"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [x] This pull request includes tests or only affects documentation.
- [x] This pull request includes a label categorizing the change e.g. `fix`, `feature`, `enhancement`

### CODE REVIEW

File: RELEASE-NOTES.md [hunk 1]
```diff
@@ -1,5 +1,42 @@
 # Prefect Release Notes
 
+## Release 2.8.4
+
+### Enhancements
+- Create `EventsWorker` to manage client lifecycle and abstract async nature — https://github.com/PrefectHQ/prefect/pull/8673
```
Comment (madkinsz) [inline RELEASE-NOTES.md:6]: experimental

File: RELEASE-NOTES.md [hunk 2]
```diff
@@ -1,5 +1,42 @@
...
+- Enable `DefaultAzureCredential` authentication for Azure filesystem block — https://github.com/PrefectHQ/prefect/pull/7513
+- Add support for yaml config strings to `KubernetesClusterConfig` — https://github.com/PrefectHQ/prefect/pull/8643
+- Add `--description` flag to `prefect deployment build` CLI command — https://github.com/PrefectHQ/prefect/pull/8603
+- Track flow run id when generating task run results — https://github.com/PrefectHQ/prefect/pull/8674
+- Handle SIGTERM received by server gracefully — https://github.com/PrefectHQ/prefect/pull/7948
+- Optimize database query performance — https://github.com/PrefectHQ/prefect/pull/8659
```
Comment (madkinsz) [inline RELEASE-NOTES.md:12]: needs more info

File: RELEASE-NOTES.md [hunk 3]
```diff
@@ -1,5 +1,42 @@
 # Prefect Release Notes
 
+## Release 2.8.4
+
+### Enhancements
+- Create `EventsWorker` to manage client lifecycle and abstract async nature — https://github.com/PrefectHQ/prefect/pull/8673
+- Enable `DefaultAzureCredential` authentication for Azure filesystem block — https://github.com/PrefectHQ/prefect/pull/7513
+- Add support for yaml config strings to `KubernetesClusterConfig` — https://github.com/PrefectHQ/prefect/pull/8643
+- Add `--description` flag to `prefect deployment build` CLI command — https://github.com/PrefectHQ/prefect/pull/8603
+- Track flow run id when generating task run results — https://github.com/PrefectHQ/prefect/pull/8674
```
Comment (madkinsz) [inline RELEASE-NOTES.md:10]: experimental

File: RELEASE-NOTES.md [hunk 4]
```diff
@@ -1,5 +1,42 @@
...
+- Handle SIGTERM received by server gracefully — https://github.com/PrefectHQ/prefect/pull/7948
+- Optimize database query performance — https://github.com/PrefectHQ/prefect/pull/8659
+- Add clarifying modal to `Task Inputs` tab in UI — https://github.com/PrefectHQ/prefect/pull/8295
+
+### Fixes
+- Fix loading of existing deployment descriptions from the server — https://github.com/PrefectHQ/prefect/pull/8675
```
Comment (madkinsz) [inline RELEASE-NOTES.md:16]: can be removed if we want, the thing being fixed was unreleased

File: RELEASE-NOTES.md [hunk 5]
```diff
@@ -1,5 +1,42 @@
...
+- Ensure flow parameters default values are present during deployment runs — https://github.com/PrefectHQ/prefect/pull/8666
+- Use a monotonic clock for Kubernetes job watch timeout deadline calculation — https://github.com/PrefectHQ/prefect/pull/8680
+- Fix version misaligned on the settings page in the UI — https://github.com/PrefectHQ/prefect/pull/8676
+
+### Experimental
+- Refactor supervisors to manage submission — https://github.com/PrefectHQ/prefect/pull/8631
+- Improve supervisor repr for debugging — https://github.com/PrefectHQ/prefect/pull/8633
+- Add timeout support to supervisors — https://github.com/PrefectHQ/prefect/pull/8649
+- Update `send_call` to `send_call_to_supervisor` — https://github.com/PrefectHQ/prefect/pull/8653
```
Comment (madkinsz) [inline RELEASE-NOTES.md:25]: let's drop this one

File: RELEASE-NOTES.md [hunk 6]
```diff
@@ -1,5 +1,42 @@
...
+- Add security headers for docs — https://github.com/PrefectHQ/prefect/pull/8655
+- Add markdown link fix in orchestration docs — https://github.com/PrefectHQ/prefect/pull/8660
+
+## New Contributors
+* @samdyzon made their first contribution in https://github.com/PrefectHQ/prefect/pull/7513
+* @mjschock made their first contribution in https://github.com/PrefectHQ/prefect/pull/8660
+* @jcorrado76 made their first contribution in https://github.com/PrefectHQ/prefect/pull/8603
+* @scharlottej13 made their first contribution in https://github.com/PrefectHQ/prefect/pull/8669
+
+**All changes**: https://github.com/PrefectHQ/prefect/compare/2.8.3...preview
```
Comment (madkinsz) [inline RELEASE-NOTES.md:38]:
```suggestion
**All changes**: https://github.com/PrefectHQ/prefect/compare/2.8.3...2.8.4
```

**Review by madkinsz — COMMENTED** [review]
Can you update the announce bar too?

Let's thank and highlight our 4 new contributors in the short summary for the GH Release and community posts.

File: RELEASE-NOTES.md [hunk 7]
```diff
@@ -1,5 +1,42 @@
...
+
+### Enhancements
+- Create `EventsWorker` to manage client lifecycle and abstract async nature — https://github.com/PrefectHQ/prefect/pull/8673
+- Enable `DefaultAzureCredential` authentication for Azure filesystem block — https://github.com/PrefectHQ/prefect/pull/7513
+- Add support for yaml config strings to `KubernetesClusterConfig` — https://github.com/PrefectHQ/prefect/pull/8643
+- Add `--description` flag to `prefect deployment build` CLI command — https://github.com/PrefectHQ/prefect/pull/8603
+- Track flow run id when generating task run results — https://github.com/PrefectHQ/prefect/pull/8674
+- Handle SIGTERM received by server gracefully — https://github.com/PrefectHQ/prefect/pull/7948
+- Optimize database query performance — https://github.com/PrefectHQ/prefect/pull/8659
+- Add clarifying modal to `Task Inputs` tab in UI — https://github.com/PrefectHQ/prefect/pull/8295
```
Comment (billpalombi) [inline RELEASE-NOTES.md:13]:
```suggestion
- Add clarifying modal to `Task Inputs` tab to the task run page in the UI — https://github.com/PrefectHQ/prefect/pull/8295
```

**Review by billpalombi — COMMENTED** [review]
One minor nit, otherwise lgtm

File: RELEASE-NOTES.md (hunk 7 above)
Comment (serinamarie) [reply RELEASE-NOTES.md:11]:
```suggestion
- Add clarifying modal to the task run page in the UI — https://github.com/PrefectHQ/prefect/pull/8295
```

**Review by rpeden — APPROVED** [review]
Looks good to me!

**Review by cicdw — APPROVED** [review]

**Pull request merged**